Sources/SPTDataLoader.m

```diff
@@ -175,10 +198,19 @@ - (void)successfulResponse:(SPTDataLoaderResponse *)response
     if (![self isRequestExpected:response.request]) {
         return;
     }
-    
-    [self executeDelegateBlock: ^{
-        [self.delegate dataLoader:self didReceiveSuccessfulResponse:response];
-    }];
+    if (self.delegate) {
+        [self executeDelegateBlock:^{
+            [self.delegate dataLoader:self didReceiveSuccessfulResponse:response];
+        }];
+    } else {
+        [self executeDelegateBlock:^{
+            void (^handler)(SPTDataLoaderResult *) = [self.requestHandlers objectForKey:response.request];
+            handler([SPTDataLoaderResult fromResponse:response kind:SPTDataLoaderResultSuccess]);
```

Comment:
Always check for `nil` before invoking a block. If `handler == nil`, this will crash.